lib/src/events.dart

```diff
@@ -244,3 +244,18 @@ class SpeakingChangedEvent with ParticipantEvent {
     required this.speaking,
   });
 }
+
+/// One of subscribed tracks have changed its permissions for the current
+/// participant. If permission was revoked, then the track will no longer
+/// be subscribed. If permission was granted, a TrackSubscribed event will
```

Comment:
Actually, track is still subscribed from server's point of view. When permission is revoked, server will keep the subscription in a pending list and resume the track when the permissions are granted (if the permissions are granted)

Reply:
Are you emitting `TrackSubscribed` on resume?

Reply:
Interesting point! I copied this comment from dz's PR https://github.com/livekit/client-sdk-js/pull/109
I might be misunderstanding something, so technically it will remain subscribed but it is desired to emit TrackSubscribed / TrackUnsubscribed ?

Reply:
It might be best if I actually test the behavior 😅

Reply:
also I was wondering if permissions can be set at publish time ? or it's only possible after publish.

Reply:
I am almost useless here @hiroshihorie . I do not fully remember what events get emitted and the exact meaning of those.

Functionally, subscription/unsubscription action is something which the app tells the SDK explicitly. The permissions is something that is coming from the publisher side. So, because of permissions, a track pausing/resuming does not constitute unsubscription as the app did not ask for it. So, the track is subscribed and server will send media when there are permissions. If there are no permissions, it is notified via `SubscriptionPermissionUpdate`.

What I am trying to say is the notion of `subscribed` is app asking our SDK to subscribe to a track. We get that message and send media if there is permission.

I am stressing on that because apps need not do anything different in the subscription path. They just subscribe and be done with it. Until they want to unsubscribe. And the media may pause/resume because of permissions and we emit the SusbcriptionPermissionUpdate to let them know. `TrackSubscribed`/`TrackUnsubscribed` events seem like it is in response to their API and I do not think we should emit them on track pause/resume because of permissions.

Reply:
Permissions is a separate message now. It needs to be because it can change in the middle.

But, good question about SDKs providing that option at creation time and SDKs converting that into a second protocol message. WDYT @davidzhao @davidliu

Reply:
I think it may be useful, but IMO adds cognitive load to `publishTrack`. I say let's wait and see how people are using it first before linking the two things together. Even with the current APIs, it's possible to achieve minimal permissions so there isn't "leakage" from new tracks being subscribed by unintended participants.

Reply:
> Interesting point! I copied this comment from dz's PR [livekit/client-sdk-js#109](https://github.com/livekit/client-sdk-js/pull/109) I might be misunderstanding something, so technically it will remain subscribed but it is desired to emit TrackSubscribed / TrackUnsubscribed ?

To the client's perspective, the track will not be available for playback, and an `Unsubscribed` event will be triggered. When permissions are granted again, a `Subscribed` event should be fired again. CC: @davidliu  as well to make sure we are aligned on behavior.

Without the explicit Unsubscribed event, the client's UI may be stuck in a weird state where they are still bound to the remote track, but will not be getting any frames.